thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DatasetTriggerInfoRepoTest.java

```diff
@@ -0,0 +1,141 @@
+        1, Collections.singletonList(TEST_DATA_SOURCE));
+    Assert.assertSame(datasetTriggerInfoRepo,
+        DatasetTriggerInfoRepo.initAndGetInstance(1, Collections.singletonList(TEST_DATA_SOURCE)));
+    Assert.assertSame(datasetTriggerInfoRepo, DatasetTriggerInfoRepo.getInstance());
+    Assert.assertTrue(datasetTriggerInfoRepo.isDatasetActive(TEST_DATASET_PREFIX + 1));
+    Assert.assertTrue(datasetTriggerInfoRepo.isDatasetActive(TEST_DATASET_PREFIX + 2));
+    Assert.assertFalse(datasetTriggerInfoRepo.isDatasetActive(TEST_DATASET_PREFIX + 3));
+    Assert.assertEquals(datasetTriggerInfoRepo.getLastUpdateTimestamp(TEST_DATASET_PREFIX + 1), 1000);
+    Assert.assertEquals(datasetTriggerInfoRepo.getLastUpdateTimestamp(TEST_DATASET_PREFIX + 2), 2000);
+    Assert.assertEquals(datasetTriggerInfoRepo.getLastUpdateTimestamp(TEST_DATASET_PREFIX + 3), 0);
+  }
+
+  @Test
+  public void testSetLastUpdateTimestamp() {
+    DatasetTriggerInfoRepo datasetTriggerInfoRepo = DatasetTriggerInfoRepo.initAndGetInstance(
+        1, Collections.singletonList(TEST_DATA_SOURCE));
+    datasetTriggerInfoRepo.setLastUpdateTimestamp(TEST_DATASET_PREFIX + 1, 3000);
+    Assert.assertEquals(datasetTriggerInfoRepo.getLastUpdateTimestamp(TEST_DATASET_PREFIX + 1), 3000);
+    Assert.assertEquals(datasetTriggerInfoRepo.getLastUpdateTimestamp(TEST_DATASET_PREFIX + 2), 2000);
+    Assert.assertEquals(datasetTriggerInfoRepo.getLastUpdateTimestamp(TEST_DATASET_PREFIX + 4), 0);
+  }
+
+  @Test
+  public void testRefresh() throws InterruptedException {
+    DetectionConfigManager detectionConfigManager = DAORegistry.getInstance().getDetectionConfigManager();
+    MetricConfigManager metricConfigManager = DAORegistry.getInstance().getMetricConfigDAO();
+    DatasetConfigManager datasetConfigDAO = DAORegistry.getInstance().getDatasetConfigDAO();
+    DatasetTriggerInfoRepo datasetTriggerInfoRepo = DatasetTriggerInfoRepo.initAndGetInstance(
+        1, Collections.singletonList(TEST_DATA_SOURCE));
+
+    MetricConfigDTO metric = new MetricConfigDTO();
+    metric.setDataset(TEST_DATASET_PREFIX + 3);
+    metric.setName(TEST_METRIC_PREFIX + 3);
+    metric.setActive(true);
+    metric.setAlias("");
+    long metricId = metricConfigManager.save(metric);
+
+    DetectionConfigDTO detect2 = new DetectionConfigDTO();
+    detect2.setName("test_detection_2");
+    detect2.setActive(true);
+    Map<String, Object> props = new HashMap<>();
+    List<String> metricUrns = new ArrayList<>();
+    metricUrns.add("thirdeye:metric:" + metricId);
+    props.put("nestedMetricUrns", metricUrns);
+    detect2.setProperties(props);
+    detectionConfigManager.save(detect2);
+
+    DatasetConfigDTO ds = new DatasetConfigDTO();
+    ds.setDataset(TEST_DATASET_PREFIX + 3);
+    ds.setDataSource(TEST_DATA_SOURCE);
+    ds.setLastRefreshTime(3000);
+    datasetConfigDAO.save(ds);
+
+    Thread.sleep(65 * 1000); // wait for datasetTriggerInfoRepo to refresh
```

Comment:
This issue is that `DatasetTriggerInfoRepo` can only refresh at minute level. When I run the test suite, it does not add that much time since the entire suite runs for couple minutes.